fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/EntityDatatableChecksWritePlatformServiceImpl.java

```diff
@@ -105,11 +104,11 @@ public CommandProcessingResult createCheck(final JsonCommand command) {
                     throw new EntityDatatableCheckAlreadyExistsException(entity, status, datatableName);
                 }
             } else {
-                EntityTables entityTable = EntityTables.fromName(entity);
-                if (EntityTables.LOAN == entityTable) {
+                EntityTables entityTable = EntityTables.fromEntityName(entity);
+                if (entityTable == EntityTables.LOAN) {
```

Comment:
So why did we flip the condition? To be honest what you did is also my preference but it's unnecessary. The result is the same yet it's adding an extra change to your PR.

Reply:
This was a conflict between me and Muthu, we were working on the same code, and I resolved this way.